src/libstore/build/local-derivation-goal.cc

```diff
@@ -803,9 +803,21 @@ void LocalDerivationGoal::startBuilder()
         userNamespaceSync.create();
 
         Path maxUserNamespaces = "/proc/sys/user/max_user_namespaces";
-        static bool userNamespacesEnabled =
-            pathExists(maxUserNamespaces)
-            && trim(readFile(maxUserNamespaces)) != "0";
+        Path procSelfNsUser = "/proc/self/ns/user";
+        static bool userNamespacesEnabled = false;
+        static bool userNamespacesTested = false;
+
+        /* Only check once */
+        if (!userNamespacesTested) {
+            if (pathExists(maxUserNamespaces)) {
+                userNamespacesEnabled = (trim(readFile(maxUserNamespaces)) != "0");
```

Comment:
Do we still need to check that or is `/proc/self/ns/user` enough?

Reply:
I think checking only for `/proc/self/ns/user` at the beginning seems reasonable. It would also solve the problem when the max namespaces initially are set to 0 and then become available again. I will modify the patch accordingly.